src/System.Private.CoreLib/shared/System/Globalization/CultureInfo.cs

```diff
@@ -1095,169 +1094,42 @@ private void VerifyWritable()
+        public static CultureInfo GetCultureInfo(int culture)
         {
-            // retval is our return value.
-            CultureInfo? retval;
-
-            // Temporary hashtable for the names.
-            Dictionary<string, CultureInfo>? tempNameHT = s_NameCachedCultures;
-
-            if (name != null)
+            if (culture <= 0)
             {
-                name = CultureData.AnsiToLower(name);
+                throw new ArgumentOutOfRangeException(nameof(culture), SR.ArgumentOutOfRange_NeedPosNum);
             }
 
-            if (altName != null)
-            {
-                altName = CultureData.AnsiToLower(altName);
-            }
+            Dictionary<int, CultureInfo> lcidTable = CachedCulturesByLcid;
+            CultureInfo? result;
 
-            // We expect the same result for both hashtables, but will test individually for added safety.
-            if (tempNameHT == null)
-            {
-                tempNameHT = new Dictionary<string, CultureInfo>();
-            }
-            else
+            lock (lcidTable)
             {
-                // If we are called by name, check if the object exists in the hashtable.  If so, return it.
-                if (lcid == -1 || lcid == 0)
+                if (lcidTable.TryGetValue(culture, out result))
                 {
-                    Debug.Assert(name != null && (lcid != -1 || altName != null));
-                    bool ret;
-                    lock (_lock)
-                    {
-                        ret = tempNameHT.TryGetValue(lcid == 0 ? name! : name! + '\xfffd' + altName!, out retval);
-                    }
-
-                    if (ret && retval != null)
-                    {
-                        return retval;
-                    }
+                    return result;
                 }
             }
 
-            // Next, the Lcid table.
-            Dictionary<int, CultureInfo>? tempLcidHT = s_LcidCachedCultures;
-
-            if (tempLcidHT == null)
-            {
-                // Case insensitive is not an issue here, save the constructor call.
-                tempLcidHT = new Dictionary<int, CultureInfo>();
-            }
-            else
-            {
-                // If we were called by Lcid, check if the object exists in the table.  If so, return it.
-                if (lcid > 0)
-                {
-                    bool ret;
-                    lock (_lock)
-                    {
-                        ret = tempLcidHT.TryGetValue(lcid, out retval);
-                    }
-                    if (ret && retval != null)
-                    {
-                        return retval;
-                    }
-                }
-            }
-
-            // We now have two temporary hashtables and the desired object was not found.
-            // We'll construct it.  We catch any exceptions from the constructor call and return null.
             try
```

Comment:
>try [](start = 12, length = 3)

We can avoid throwing and catching here if we added a extra default bool param to the CultureData method:

internal static CultureData GetCultureData(int culture, bool bUseUserOverride)

So we can avoid having it throw and then use 
new CultureInfo(cultureData); 

This is not worth it to do but I wanted to pint out it is possible.